SpatialGDK/Source/SpatialGDK/Public/Schema/UnrealMetadata.h

```diff
@@ -76,19 +79,20 @@ struct UnrealMetadata : Component
 #if !UE_BUILD_SHIPPING
 		if (NativeClass.IsStale())
 		{
-			UE_LOG(LogSpatialClassInfoManager, Warning, TEXT("UnrealMetadata native class %s unloaded whilst entity in view."), *ClassPath);
+			UE_LOG(LogSpatialUnrealMetadata, Warning, TEXT("UnrealMetadata native class %s unloaded whilst entity in view."), *ClassPath);
 		}
 #endif
-		UClass* Class = nullptr;
+		UClass* Class = FindObject<UClass>(nullptr, *ClassPath, false);
 
 		// Unfortunately StablyNameRef doesn't mean NameStableForNetworking as we add a StablyNameRef for every startup actor (see USpatialSender::CreateEntity)
 		// TODO: UNR-2537 Investigate why FindObject can be used the first time the actor comes into view for a client but not subsequent loads.
-		if (StablyNamedRef.IsSet() && bNetStartup.IsSet() && bNetStartup.GetValue())
-		{
-			Class = FindObject<UClass>(nullptr, *ClassPath, false);
-		}
-		else
+		if (Class == nullptr && !(StablyNamedRef.IsSet() && bNetStartup.IsSet() && bNetStartup.GetValue()))
 		{
+			if (GetDefault<USpatialGDKSettings>()->bAsyncLoadNewClassesOnEntityCheckout)
+			{
+				UE_LOG(LogSpatialUnrealMetadata, Warning, TEXT("Class %s couldn't be found even though async loading on entity checkout is enabled. Will attempt to load it synchronously."), *ClassPath);
```

Comment:
Same preference point as before - prefer a literal main body then variable data at the end.